src/porepy/grids/md_grid.py

```diff
@@ -396,6 +466,13 @@ def add_subdomains(self, new_subdomains: Union[pp.Grid, Iterable[pp.Grid]]) -> N
             # Add the grid to the dictionary of subdomains with an empty data dictionary.
             self._subdomain_data[sd] = {}
 
+        # Generate boundary grids for each subdomain.
```

Comment:
@keileg, I believe you worked on this recently. Please have a look. Note that BoundaryGrid does not inherit from Grid anymore.

Reply:
I actually moved the BoundaryGrid construction out of this method, because of the issue @Yuriyzabegaev hints to regarding splitting of faces when fracture intersections are identified. I agree that doing the update inside `MixedDimensionalGrid` is more natural (hence my original placement of the BC-grid generation was the same as suggested here), but I am not convinced about the solution of adding it to geometry computation:
1. The purist case, this is topology, not geometry.
2. More fundamentally, this is not an operation I would expect from a computation of geometry, it breaks the single responsibility principle.
Is it an option to have a separate method `mdg.update_boundary_topology()` or similar, and call this after having split the faces?

Reply:
Added the separate method.